### PR TITLE
[feature] ZZPhase support for devices that support it

### PR DESCRIPTION
**Pull request by lmondada**
As I currently don't have access to any gate fidelity information for the devices, I assume perfect fidelity. This can obviously easily be changed in the future.

Note that I removed the `ZZPhaseToRz` pass when `FullPeephole` is run, as the KAK should already do the optimal thing.

### CODE REVIEW

**Review by sjdilkes — DISMISSED** [review]

**Comment by lmondada** [comment]
Aaargh right, this won't pass until we merge and release https://github.com/CQCL/tket/pull/476. We should probably make a pytket 1.5.1 release for this.

**Comment by cqc-alec** [comment]
I re-ran with pytket 1.5.1, but now some predicate checks are failing in the tests.

**Comment by cqc-alec** [comment]
Pytket 1.5.2 is now released. Could you update the requirement in `setup.py` to `"pytket ~= 1.5, >= 1.5.2"`?

File: setup.py [hunk 1]
```diff
@@ -39,7 +39,7 @@
     packages=find_namespace_packages(include=["pytket.*"]),
     include_package_data=True,
     install_requires=[
-        "pytket ~= 1.5",
+        "pytket ~= 1.5.2",
```
Comment (cqc-alec) [inline setup.py:42]: This will restrict to 1.5.x. I think it should be `~= 1.5, >= 1.5.2`, That way 1.6 will be allowed.

Comment (cqc-alec) [reply setup.py:42]: (TBH I'm not completely sure, but I think that's how it works.)

**Comment by cqc-alec** [comment]
I removed `test_zz_phase_support()` because it does not work at optimization level 1 (as I found by hacking the backend object to always support `ZZPhase`). However at optimization level 2 we get the expected result.

**Comment by cqc-alec** [comment]
This now LGTM but let's hold back from merging until we have understood the platform-dependency issue.

**Comment by cqc-alec** [comment]
See https://github.com/CQCL/tket/issues/488 for the underlying issue in TKET. I think this can be merged anyway.

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**